File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int>st(nums.begin(),nums.end());
        while(st.find(head->val) != st.end()) head = head->next;
        ListNode* prev = head;
        ListNode* temp = head->next;
        while(temp)
        {
            if(st.find(temp->val) != st.end())
            {
                prev->next = temp->next;
                temp = temp->next;
            }
            else
            {
                prev = prev->next;
                temp = temp->next;
            }
        }
        return head;
    }
};

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        HashSet<Integer> set = new HashSet<>();
        for(int num : nums) {
            set.add(num);
        }
        ListNode prev = new ListNode();
        ListNode start = prev;
        while(head != null) {
            int val = head.val;
            if(set.contains(val)) {
                head = head.next;
                continue;
            }
            prev.next = head;
            prev = prev.next;
            head = head.next;
        }
        prev.next = null;
        return start.next;
    }
}